src/sfizz/Synth.h

```diff
@@ -668,15 +695,25 @@ class Synth final : public Voice::StateListener, public Parser::Listener {
     // Default active switch if multiple keyswitchable regions are present
     absl::optional<uint8_t> defaultSwitch;
     std::vector<std::string> unknownOpcodes;
-    using RegionPtrVector = std::vector<Region*>;
-    using VoicePtrVector = std::vector<Voice*>;
-    std::vector<std::unique_ptr<Region>> regions;
-    std::vector<std::unique_ptr<Voice>> voices;
+    using RegionViewVector = std::vector<Region*>;
+    using VoiceViewVector = std::vector<Voice*>;
+    using VoicePtr = std::unique_ptr<Voice>;
+    using RegionPtr = std::unique_ptr<Region>;
+    using RegionSetPtr = std::unique_ptr<RegionSet>;
+    std::vector<RegionPtr> regions;
+    std::vector<VoicePtr> voices;
+    // These are more general "groups" than sfz and encapsulates the full hierarchy
+    enum class Header { Global, Master, Group };
```

Comment:
This exists already as OpcodeScope found in `Opcode.h`.

Reply:
OK. I added the `master` level. It seems there's no need to handle this anywhere else in the regex generation but can you check?

Reply:
It's not affected, it would be always called with region scope, after all of master/group/region are coalesced into one big array.